src/blackd/__init__.py

```diff
@@ -146,11 +146,13 @@ async def handle(request: web.Request, executor: Executor) -> web.Response:
         )
 
         # Preserve CRLF line endings
-        if req_str[req_str.find("\n") - 1] == "\r":
+        nl = req_str.find("\n")
+        if nl > 0 and req_str[nl - 1] == "\r":
             formatted_str = formatted_str.replace("\n", "\r\n")
             # If, after swapping line endings, nothing changed, then say so
             if formatted_str == req_str:
                 raise black.NothingChanged
+        del nl
```

Comment:
This is a bit of a premature optimisation, isn't it?

Reply:
Oh okay, I was just deleting it so there wasn't random variables floating around, I'll remove it.